python/ray/experimental/client/__init__.py

```diff
@@ -1,148 +1,108 @@
-# * _server_api should be set if we're inside the clientserver
-# * Both will be set if we're running both (as in a test)
-# * Neither should be set if we're inside the raylet (but we still need to shim
-#       from the client API surface to the Ray API)
-#
-# The job of RayAPIStub (below) delegates to the appropriate one of these
-# depending on what's set or not. Then, all users importing the ray object
-# from this package get the stub which routes them to the appropriate APIImpl.
-_client_api: Optional[APIImpl] = None
-_server_api: Optional[APIImpl] = None
-
-# The reason for _is_server is a hack around the above comment while running
-# tests. If we have both a client and a server trying to control these static
-# variables then we need a way to decide which to use. In this case, both
-# _client_api and _server_api are set.
-# This boolean flips between the two
-_is_server: bool = False
-
-
-@contextmanager
-def stash_api_for_tests(in_test: bool):
-    global _is_server
-    is_server = _is_server
-    if in_test:
-        _is_server = True
-    try:
-        yield _server_api
-    finally:
-        if in_test:
-            _is_server = is_server
-
-
-def _set_client_api(val: Optional[APIImpl]):
-    global _client_api
-    global _is_server
-    if _client_api is not None:
-        raise Exception("Trying to set more than one client API")
-    _client_api = val
-    _is_server = False
-
-
-def _set_server_api(val: Optional[APIImpl]):
-    global _server_api
-    global _is_server
-    if _server_api is not None:
-        raise Exception("Trying to set more than one server API")
-    _server_api = val
-    _is_server = True
-
-
-def reset_api():
-    global _client_api
-    global _server_api
-    global _is_server
-    _client_api = None
-    _server_api = None
-    _is_server = False
-
-
-def _get_client_api() -> APIImpl:
-    global _client_api
-    return _client_api
-
-
-def _get_server_instance():
-    """Used inside tests to inspect the running server.
+
+class RayAPIStub:
     """
-    global _server_api
-    if _server_api is not None:
-        return _server_api.server
+    This class stands in as the replacement API for the `import ray` module.
 
+    Much like the ray module, this mostly delegates the work to the
+    _client_worker. As parts of the ray API are covered, they are piped through
+    here or on the client worker API.
+    """
+
+    def __init__(self):
+        from ray.experimental.client.api import ClientAPI
+        self.api = ClientAPI()
+        self.client_worker = None
+        self._server = None
+        self._connected_with_init = False
+        self._inside_client_test = False
 
-class RayAPIStub:
     def connect(self,
                 conn_str: str,
                 secure: bool = False,
-                metadata: List[Tuple[str, str]] = None,
-                stub=None) -> None:
+                metadata: List[Tuple[str, str]] = None) -> None:
+        """
```

Comment:
Style nits: no newline (`"""Connect the Ray Client to a server`) here and elsewhere.